### PR TITLE
Adds the ability to get debug info from the backburner instance.

### PR DESCRIPTION
**Pull request by scalvert**
Adds the ability to get debug info when the backburner instance's `DEBUG` flag is set to `true`. This internally leverages the existing functionality of retrieving stacktraces from queues. 

Changes include:
- a new `public` method on the `Backburner` class - `getDebugInfo`
- a new `public` but privileged `_getDebugInfo` method on the `DeferredActionQueues` class
- a new `public` but privileged method on the `Queue` class

The intent is that this can be leveraged for additional information when identifying timer leaks within ember-qunit, which already contains some [leak detection](https://github.com/emberjs/ember-qunit/blob/master/addon-test-support/ember-qunit/test-isolation-validation.js).

Sample output:

```
{
  "counters": {
    "begin": 59,
    "end": 49,
    "events": {
      "begin": 58,
      "end": 0
    },
    "autoruns": {
      "created": 13,
      "completed": 12
    },
    "run": 16,
    "join": 0,
    "defer": 0,
    "schedule": 17,
    "scheduleIterable": 0,
    "deferOnce": 0,
    "scheduleOnce": 10,
    "setTimeout": 1,
    "later": 13,
    "throttle": 4,
    "debounce": 60,
    "cancelTimers": 2,
    "cancel": 35,
    "loops": {
      "total": 58,
      "nested": 2
    }
  },
  "timers": [],
  "instanceStack": [
    {
      "one": [],
      "two": [],
      "three": [
        {
          "target": null,
          "stack": "Three stack"
        }
      ],
      "four": [
        {
          "target": null,
          "stack": "Four stack"
        }
      ]
    },
    {
      "one": [
        {
          "target": null,
          "stack": "One stack"
        }
      ],
      "two": [
        {
          "target": null,
          "stack": "Two stack"
        }
      ],
      "three": [],
      "four": []
    }
  ]
}
```

TODO:
- [x] add more tests for autorun scenarios
- [x] add more tests for nested runs
- [x] add tests for main APIs (later, debounce, etc)
- [x] add docs to README
- [x] add examples of returned debug info to this PR description

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -546,6 +546,18 @@ export default class Backburner {
     this._ensureInstance();
   }
 
+  public getDebugInfo() {
+    if (this.DEBUG) {
+      return {
+        counters: this.counters,
```
Comment (scalvert) [inline lib/index.ts]: I'm not convinced we need this. @rwjblue - what are you thinking this would be good for?

File: lib/backburner/queue.ts [hunk 2]
```diff
@@ -169,6 +176,30 @@ export default class Queue {
...
+      let debugInfo: Array<IQueueItem> = new Array<IQueueItem>();
+
+      for (let i = 0; i < queueItems.length; i += 4) {
+        let maybeStack: Error | undefined = queueItems[i + 3];
+
+        let queueItem = {
+          target: queueItems[i],
+          method: queueItems[i + 1],
+          args: queueItems[i + 2],
+          stack: typeof maybeStack !== 'undefined' ? maybeStack.stack : ''
```
Comment (stefanpenner) [inline lib/backburner/queue.ts]: `maybeStack !== null && typeof mabyeStack === 'object'` ? maybeStack.stack : null

File: lib/backburner/utils.ts [hunk 3]
```diff
@@ -1,5 +1,16 @@
+import { IQueueItem } from './interfaces';
+
+type MaybeError = Error | undefined;
+
 const NUMBER = /\d+/;
 
+enum QueueItemPosition {
```
Comment (rwjblue) [inline lib/backburner/utils.ts]: Can you change this to a `const enum`?

Comment (scalvert) [reply lib/backburner/utils.ts]: yep!

**Comment by scalvert** [comment]
@rwjblue - what else do you think this PR needs?

**Pull request merged**